src/sparse/KokkosSparse_CrsMatrix.hpp

```diff
@@ -554,7 +554,37 @@ class CrsMatrix {
              OrdinalType* rowmap,
              OrdinalType* cols)
   {
-    ctor_impl (label, nrows, ncols, annz, val, rowmap, cols);
+    using Kokkos::Unmanaged;
+    using HostRowmap = Kokkos::View<SizeType*, Kokkos::HostSpace>;
+    using UnmanagedRowmap = Kokkos::View<const SizeType*, Kokkos::HostSpace, Kokkos::MemoryTraits<Unmanaged>>;
+    using UnmanagedEntries = Kokkos::View<const OrdinalType*, Kokkos::HostSpace, Kokkos::MemoryTraits<Unmanaged>>;
+    using UnmanagedValues = Kokkos::View<const ScalarType*, Kokkos::HostSpace, Kokkos::MemoryTraits<Unmanaged>>;
+    //Allocate device rowmap, entries, values views
+    typename row_map_type::non_const_type rowmapDevice(Kokkos::ViewAllocateWithoutInitializing("rowmap"), nrows + 1);
+    index_type entriesDevice(Kokkos::ViewAllocateWithoutInitializing("entries"), annz);
+    //given rowmap in ordinal_type, so may need to convert to size_type explicitly
+    HostRowmap rowmapConverted;
+    UnmanagedRowmap rowmapRaw;
+    if(!std::is_same<OrdinalType, SizeType>::value)
+    {
+      rowmapConverted = HostRowmap(Kokkos::ViewAllocateWithoutInitializing("rowmap raw"), nrows + 1);
+      for(OrdinalType i = 0; i <= nrows; i++)
+        rowmapConverted(i) = rowmap[i];
+      rowmapRaw = rowmapConverted;
+    }
+    else
+    {
+      rowmapRaw = UnmanagedRowmap((const SizeType*) rowmap, nrows + 1);
+    }
+    Kokkos::deep_copy(rowmapDevice, rowmapRaw);
+    UnmanagedEntries entriesRaw(cols, annz);
+    Kokkos::deep_copy(entriesDevice, entriesRaw);
+    //Construct graph and populate all members
+    this->numCols_ = ncols;
+    this->graph = StaticCrsGraphType(entriesDevice, rowmapDevice);
```

Comment:
Is a fence needed between `deep_copy` and `StaticCrsGraph` construction?

Reply:
@e10harvey No, the StaticCrsGraph constructor doesn't have to access any data in the views, it just does shallow copies. But if it did in the future, the deep_copy fences.